packages/mina-service/src/service/task-service.ts

```diff
@@ -28,29 +42,52 @@ export class TaskService {
         break;
       }
 
-      const backoff = await withCompoundTransaction(async (session) => {
-        const task = await Proof.getNextTask(session);
-        if (task === null) {
-          return true;
-        }
+      let backoff = true;
 
-        logger.debug('Task received:', task);
+      const task = await withCompoundTransaction(async (session) =>
+        ModelQueueTask.getInstance().acquireNextTaskInQueue(
+          session.proofService
+        )
+      );
 
-        try {
-          await Proof.create(task, session);
-          return false;
-        } catch (error) {
-          logger.error(`Error processing task with ID ${task._id}: ${error}`);
-          return true;
-        }
-      });
+      if (task !== null) {
+        backoff = await withCompoundTransaction(async (session) => {
+          logger.debug('Task received:', task);
```

Comment:
if only use transaction from proof db, you should use withTransaction(fn, "proofService") instead

Reply:
prover needs a compound session, for example, to fetch document metadata:
https://github.com/orochi-network/zkDatabase/blob/a0d5fb1b226d4219024eaa77462c736786b8b637/packages/mina-service/src/domain/proof.ts#L41-L47

Reply:
I agree, Closed